File: app/src/main/java/com/thanksmister/iot/mqtt/alarmpanel/ui/adapters/ForecastCardAdapter.java
package com.thanksmister.iot.mqtt.alarmpanel.ui.adapters;

import android.content.Context;
import android.support.v4.content.res.ResourcesCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.thanksmister.iot.mqtt.alarmpanel.R;
import com.thanksmister.iot.mqtt.alarmpanel.network.model.Datum;
import com.thanksmister.iot.mqtt.alarmpanel.utils.DateUtils;
import com.thanksmister.iot.mqtt.alarmpanel.utils.WeatherUtils;

import java.util.List;

import butterknife.Bind;
import butterknife.ButterKnife;

public class ForecastCardAdapter extends RecyclerView.Adapter<ForecastCardAdapter.ViewHolder> {
    
    private List<Datum> items;
    private Context context;
    private String units;

    public ForecastCardAdapter(Context context, List<Datum> items, String units) {
        this.context = context;
        this.items = items;
        this.units = units;
    }
    
    // Create new views (invoked by the layout manager)
    @Override
    public ForecastCardAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemLayoutView = LayoutInflater.from(context).inflate(viewType, parent, false);
        return new ItemViewHolder(itemLayoutView);
    }

    @Override
    public int getItemViewType(int position) {
        return R.layout.adapter_forcast_card;
    }

    @Override
    public int getItemCount() {
        if (items == null) return 0;
        return items.size() > 0? items.size():0;
    }

    // Replace the contents of a view (invoked by the layout manager)
    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int position) {
        if (viewHolder instanceof ItemViewHolder) {
            Datum datum = items.get(position);
            ((ItemViewHolder) viewHolder).iconImage.setImageDrawable(ResourcesCompat.getDrawable(context.getResources(), 
                    WeatherUtils.getIconForWeatherCondition(datum.getIcon()), context.getTheme()));
            ((ItemViewHolder) viewHolder).dayText.setText(DateUtils.dayOfWeek(datum.getTime()));
            ((ItemViewHolder) viewHolder).outlookText.setText(datum.getSummary());

            //String displayUnits = (units.equals(DarkSkyRequest.UNITS_US)? context.getString(R.string.text_f): context.getString(R.string.text_c));
            String highTemp = String.valueOf(Math.round(datum.getApparentTemperatureMax()));
            String lowTemp = String.valueOf(Math.round(datum.getApparentTemperatureMin()));
            ((ItemViewHolder) viewHolder).temperatureText.setText(context.getString(R.string.text_temperature_range, highTemp, lowTemp));
        }
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

    class ItemViewHolder extends ViewHolder {
        @Bind(R.id.iconImage)
        ImageView iconImage;

        @Bind(R.id.dayText)
        TextView dayText;

        @Bind(R.id.temperatureText)
        TextView temperatureText;

        @Bind(R.id.outlookText)
        TextView outlookText;

        ItemViewHolder(View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/thanksmister/iot/mqtt/alarmpanel/ui/adapters/ForecastCardAdapter.kt
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.thanksmister.iot.mqtt.alarmpanel.ui.adapters

import android.support.v4.content.res.ResourcesCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.thanksmister.iot.mqtt.alarmpanel.R
import com.thanksmister.iot.mqtt.alarmpanel.ui.views.ForecastDisplay
import com.thanksmister.iot.mqtt.alarmpanel.utils.WeatherUtils

import kotlinx.android.synthetic.main.adapter_forcast_card.view.*

class ForecastCardAdapter(private val items: List<ForecastDisplay>?) : RecyclerView.Adapter<ForecastCardAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ForecastCardAdapter.ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.adapter_forcast_card, parent, false)
        return ViewHolder(v)
    }

    override fun getItemCount(): Int {
        if (items == null) return 0
        return if (items.isNotEmpty()) items.size else 0
    }

    override fun onBindViewHolder(holder: ForecastCardAdapter.ViewHolder, position: Int) {
        holder.bindItems(items!![position])
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bindItems(forecast: ForecastDisplay) {
            val highTemp = Math.round(forecast.temperatureHigh).toString()
            val lowTemp = Math.round(forecast.temperatureLow).toString()
            itemView.temperatureText.text = itemView.context.getString(R.string.text_temperature_range, highTemp, lowTemp)
            itemView.dayText.text = forecast.day
            itemView.outlookText.text = forecast.condition
            itemView.iconImage.setImageDrawable(ResourcesCompat.getDrawable(itemView.context.resources, forecast.conditionImage, itemView.context.theme))
        }
    }
}